File: News_Service_Jooq/src/main/java/com/dunice/projectNews/mapper/UserMapper.java
package com.dunice.projectNews.mapper;

import com.dunice.projectNews.dto.request.PutUserRequest;
import com.dunice.projectNews.dto.request.RegisterUserRequest;
import com.dunice.projectNews.dto.response.LoginUserResponse;
import com.dunice.projectNews.dto.response.PublicUserResponse;
import com.dunice.projectNews.models.User;
import com.dunice.projectNews.models.tables.records.UsersRecord;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingConstants;
import java.util.List;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
public interface UserMapper {

    User registerUserRequestToUser(RegisterUserRequest registerUserDtoRequest);

    @Mapping(target = "id", source = "userId")
    LoginUserResponse userToLoginUserResponse(User user);

    List<PublicUserResponse> listUserToListPublicUserResponse(List<User> user);

    @Mapping(target = "id", source = "userId")
    PublicUserResponse userToPublicUserResponse(User user);


    User putUserRequestToUser(PutUserRequest publicUserView);

    @Mapping(target = "userId", source ="usersId")
    User userRecordToUser(UsersRecord userRecord);
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/mapper/UserMapper.kt
package com.dunice.advancedServiceKotlin.mapper

import com.dunice.advancedServiceKotlin.dto.request.PutUserRequest
import com.dunice.advancedServiceKotlin.dto.request.RegisterUserRequest
import com.dunice.advancedServiceKotlin.dto.response.LoginUserResponse
import com.dunice.advancedServiceKotlin.dto.response.PublicUserResponse
import com.dunice.advancedServiceKotlin.models.User
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingConstants

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
interface UserMapper {

    fun registerUserRequestToUser(registerUserDtoRequest: RegisterUserRequest): User

    @Mapping(target = "id", source = "userId")
    fun userToLoginUserResponse(user: User): LoginUserResponse

    fun listUserToListPublicUserResponse(user: List<User>): List<PublicUserResponse>

    @Mapping(target = "id", source = "userId")
    fun userToPublicUserResponse(user: User): PublicUserResponse

    fun putUserRequestToUser(publicUserView: PutUserRequest): User
}
